activerecord/lib/active_record/connection_adapters/abstract/query_comment.rb

```diff
@@ -0,0 +1,143 @@
+# frozen_string_literal: true
+
+require "active_support/core_ext/module/attribute_accessors_per_thread"
+
+module ActiveRecord
+  module ConnectionAdapters
+    # SQL query comments set at the adapter level
+    module QueryComment
+      mattr_accessor :prepend_comment, default: false
+      def add_query_comment?
+        return false unless ActiveRecord::Base.query_comment_enabled
+        return false if Context.components.empty?
+        true
+      end
+
+      def add_query_comment_to_sql(sql)
+        return sql unless add_query_comment?
+        Context.set(adapter: self)
```

Comment:
Does marginalia also call this an adapter? Looking at how this is used I think we should rename this to `connection` because it's not an adapter if we can get the `pool` off of it. Although, if the only thing we need to access off of the pool is the `db_config` then we should store the db config directly rather than the whole connection.

Reply:
I've changed this to use `ActiveRecord::Base.connection_db_config`, which sounds like the correct approach and also removes the need to update the context for every call (thus invalidating the cache) - double win!